### ISSUE TITLE
Question about flow_grpo on Flux for OCR tasks and flow_grpo_fast on Flux

### ISSUE THREAD

**Issue by CIawevy** [yifan123/flow_grpo#110]
Hi! Flow_grpo 是一项出色的工作，我对强化学习了解不深，有几个问题请教：

1.目前仅 SD3 支持 flow-grpo-fast 代码，请问 flux 是否可能支持 fast 版本？若可以，代码迁移需哪些步骤？

2.关于 Flux 的 flow_grpo 训练：目前 grpo.py 未实现 OCR 任务，此前 issue 提到 OCR 任务中 Beta 是防止 reward hacking 的重要策略。我尝试设置 beta 为 0.04 时训练崩溃（如图），设为 0 则无问题。是否应设置更小的 beta 值？另外注意到 flux-kontext 的 pickscore 任务中 beta 均为 0，若训练 OCR 任务该如何设置？

3.关于 flow_grpo 的 step：从已有 issue 了解到 step 需在训练收敛后手动停止，如何公平比较不同 reward 和模型选取合适的停止条件？

<img width="1080" height="716" alt="Image" src="https://github.com/user-attachments/assets/93f1a8c9-1d87-45a4-b213-f46871d91675" />

<img width="987" height="211" alt="Image" src="https://github.com/user-attachments/assets/d2632cf1-90e0-4b4a-ba73-10a50547b5e5" />

Thanks！

**Comment by CIawevy** [comment]
关于问题2：
尝试使用上图config（beta=0.04）以后训练报错如下：

<img width="1098" height="808" alt="Image" src="https://github.com/user-attachments/assets/b128701b-9f97-4bc4-9a40-909069bbedba" />

如果beta设置为0则没有这个问题，reward快速上升
因此怀疑是beta设置太大了，这个参数是只能根据经验调整吗

**Comment by CIawevy** [comment]
> 非常感谢关注！
> 
> 1. 支持fast非常容易，改几行就行，加入window，在window内sde,window外设置noise_level=0变成ode，比如flux-kontext可以这么改
> 
> self.scheduler.set_begin_index(0)
>     with self.progress_bar(total=num_inference_steps) as progress_bar:
>         for i, t in enumerate(timesteps):
>             if i < sde_window[0]:
>                 cur_noise_level = 0
>             elif i == sde_window[0]:
>                 cur_noise_level= noise_level
>                 all_latents.append(latents)
>             elif i > sde_window[0] and i < sde_window[1]:
>                 cur_noise_level = noise_level
>             else:
>                 cur_noise_level= 0
>             self._current_timestep = t
>             latent_model_input = latents
>             if image_latents is not None:
>                 latent_model_input = torch.cat([latents, image_latents], dim=1)
>             timestep = t.expand(latents.shape[0]).to(latents.dtype)
>             noise_pred = self.transformer(
>                 hidden_states=latent_model_input,
>                 timestep=timestep / 1000,
>                 guidance=guidance,
>                 pooled_projections=pooled_prompt_embeds,
>                 encoder_hidden_states=prompt_embeds,
>                 txt_ids=text_ids,
>                 img_ids=latent_ids,
>                 joint_attention_kwargs=self.joint_attention_kwargs,
>                 return_dict=False,
>             )[0]
>             noise_pred = noise_pred[:, : latents.size(1)]
> 
>             #noise_pred = noise_pred.to(prompt_embeds.dtype)
>             latents_dtype = latents.dtype
> 
>             latents, log_prob, prev_latents_mean, std_dev_t = sde_step_with_logprob(
>                 self.scheduler, 
>                 noise_pred.float(), 
>                 t.unsqueeze(0).repeat(latents.shape[0]), 
>                 latents.float(),
>                 noise_level=cur_noise_level,
>             )
>             
>             if latents.dtype != latents_dtype:
>                 latents = latents.to(latents_dtype)
>             if i >= sde_window[0] and i < sde_window[1]:
>                 all_latents.append(latents)
>                 all_log_probs.append(log_prob)
> 2. 好像没看到训练图
> 3. 我一般是看实际效果

这里的SDE_window 如何设置呢？采用和sd3 fast一样的随机步数吗？

**Comment by yifan123** [comment]
SDE_window=2,然后窗口在num_steps的一半

**Comment by CIawevy** [comment]
thanks a lot！
